src/qibolab/transpilers/abstract.py

```diff
@@ -1,12 +1,57 @@
 from abc import ABC, abstractmethod
 from typing import List, Optional, Tuple
 
+import networkx as nx
 import numpy as np
 from qibo.backends import NumpyBackend
-from qibo.config import log
+from qibo.config import log, raise_error
 from qibo.models import Circuit
 
 
+# TODO update circuit representation
+def create_circuit_repr(circuit):
+    """Translate qibo circuit into a list of two qubit gates to be used by the transpiler.
+
+    Args:
+        circuit (:class:`qibo.models.Circuit`): circuit to be transpiled.
+
+    Returns:
+        translated_circuit (list): list containing qubits targeted by two qubit gates
+    """
+    translated_circuit = []
+    index = 0
+    for gate in circuit.queue:
+        if len(gate.qubits) == 2:
+            gate_qubits = list(gate.qubits)
+            gate_qubits.sort()
+            gate_qubits.append(index)
+            translated_circuit.append(gate_qubits)
+            index += 1
```

Comment:
In the end, you will have something like:
```py
[(x, x, 0), (x, x, 1), (x, x, 2), ...]
```

But the index can be fully inferred by the position inside `translated_circuit`:
```py
assert all(i == qs[2] for i, qs in enumerate(translated_circuit))
```

To avoid duplicating this information, just simplify the code and avoid `index`:
```suggestion
            translated_circuit.append(gate_qubits)
```

Reply:
That is true that now this can be avoided, however in the code where gate permutations are considered i need to keep track of the original position of the two qubit gates in order to be able to reconstruct the final circuit.

Reply:
What I'm saying is that the information is already there, it's the position in the list, so you don't need to add once more.

I know you need to keep before permuting, but then you can add before permutation.

It is not wrong to have it there, but don't be too eager in adding it, just do only when it's required.

(Moreover, the index is at a different level than the qubit indices, so it is better to have `((q0, q1), idx)` rather than `(q0, q1, idx)`, as proven by the suggestion below - and this is exactly what `enumerate()` does, and it is built-in)